### PR TITLE
Add mirage-block-unix.2.8.4

### PR DESCRIPTION
**Pull request by djs55**
Signed-off-by: David Scott <dave@recoil.org>

### CODE REVIEW

File: packages/mirage-block-unix/mirage-block-unix.2.8.4/opam [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "rresult"
+  "io-page-unix" {>= "2.0.0"}
+  "uri" {>= "1.9.0"}
+  "logs"
+  "ounit" {test}
+  "fmt" {test}
+]
+available: [ ocaml-version >= "4.03.0" ]
+depexts: [
+ [["alpine"] ["linux-headers"]]
```
Comment (jpdeplaix) [inline packages/mirage-block-unix/mirage-block-unix.2.8.4/opam]: Is there a reason why this is required only on alpine ? Is this package only available on Linux ? Could it depend on the ```conf-linux-libc-dev``` package instead ?

Comment (djs55) [reply packages/mirage-block-unix/mirage-block-unix.2.8.4/opam]: The dependency has only been discovered on alpine so far :) Unfortunately this package also works on Darwin and FreeBSD, so I can't use `conf-linux-libc-dev` directly. Perhaps I should copy the other `depexts` i.e.
```
  [["debian"] ["linux-libc-dev" ]]
  [["ubuntu"] ["linux-libc-dev"]]
  [["opensuse"] ["linux-glibc-devel"]]
  [["rhel"] ["kernel-headers"]]
  [["centos"] ["kernel-headers"]]
```
(I suspect these are pre-installed on the Docker images I'm using for testing, which is why I haven't spotted the problem) WDYT?

Comment (jpdeplaix) [reply packages/mirage-block-unix/mirage-block-unix.2.8.4/opam]: You could depend over ```"conf-linux-libc-dev" {os = "linux"}```. That would do the trick I suppose.

Comment (djs55) [reply packages/mirage-block-unix/mirage-block-unix.2.8.4/opam]: OK, I'll try that -- thanks!

Comment (djs55) [reply packages/mirage-block-unix/mirage-block-unix.2.8.4/opam]: I made a [test PR](https://github.com/mirage/mirage-block-unix/pull/84/files) and it didn't work unfortunately :( (at least with current `opam` 1.2). The CI failures look like this:
```
Errors found in /repo/mirage-block-unix.opam:
            error  2: File format error: at /repo/mirage-block-unix.opam:30:28,
              Expected a list of constraints
```
and
```
[WARNING] At
          C:/cygwin64/home/appveyor/.opam/4.03.0+mingw64c/packages.dev/mirage-block-unix/mirage-block-unix.opam:30:28:
          In opam-file, field 'depends:', expected pkg-version-constraints
```

- https://ci.appveyor.com/project/djs55/mirage-block-unix/build/1.0.126
- https://travis-ci.org/mirage/mirage-block-unix/builds/307510135

For now I'll add the `depexts` from `conf-linux-libc-dev` as a workaround.

**Comment by hannesm** [comment]
thanks!

**Pull request merged**